### ISSUE TITLE
Changes to `warlock.model_factory`'s signature break existing code.

### ISSUE THREAD

**Issue by neganp** [bcwaldon/warlock#25]
Changes made in commit 64a771d151a5defb9bca82042f2c78d513b33aa9 altered the signature of `warlock.model_factory` in a non backwards-compatible way that might surprise existing users.

The call

```
warlock.model_factory(schema, x)
```

executes in version 1.2 with

```
kwargs == {
    'base_class': x,
    'name': None,
}
```

but in version 1.3 it has

```
kwargs == {
    'resolver': snd_pos_arg,
    'base_class'=model.Model,
    'name': None,
}
```

This can eventually cause a base class to be passed to `jsonschema` as a resolver with predictably bad (and kind of confusing 😮 ) results. If I understand SemVer correctly, an API change that breaks existing code should come with a Major Version bump.

I've submitted a PR ( #24) which restores the original behaviour while preserving the handy new functionality.

**Comment by neganp** [comment]
New PR #39 has been submitted which is rebased on an up-to-date master.

**Comment by janw** [comment]
Hey, thanks for reopening/rebasing the PR. I've merged the fix just now, and it will be in the 1.4 release I'm planning to release soon.
